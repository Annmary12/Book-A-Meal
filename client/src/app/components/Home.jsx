import React, { Component } from 'react';
import '../assets/css/style.css';
import '../assets/css/index.css';

const Home = () => (
  <div>
      <div className="home">
    <nav className="">
    <div className="title-area">
      <a href="index.html" className="title">
        <img className="logo" src="img/logo-white.svg" width="30" />
        <span className="app-name">Book-A-Meal</span>
      </a>
      <button id="nav-toggle" href="#">&#9776;</button>
    </div>
    <div className="nav-menus">
      <ul className="nav-list">
        <li><a href="meals-caterer.html">Caterers</a></li>
        <li><a href="login.html">Login</a></li>
        <li><a href="signup.html">Sign Up</a></li>
      </ul>
    </div>
  </nav>
    <div className="mask">
      <div className="intro-text">
        <span style={{fontFamily: 'cursive'}} className="hungry">
          <span style={{fontSize: 68}}>
            <b>
              <i className="z">H</i>
              <i className="z">u</i>
              <i className="z">n</i>
              <i className="z">g</i>
              <i className="z">r</i>
              <i className="z">y</i>
              <i className="z">!</i>
              <i className="z">?,</i>
            </b>
          </span><br />
          book a meal on the go...
        </span>
      </div>
    </div>
  </div>
  <div className="content">
    <div className="menu-peak">
      <span>~~  </span><span><b>Peep into today's menu</b></span><span>  ~~</span>
    </div>
    <div className="meals">
      <div className="card">
          <div className="card-img">
            <img src="img/pounded-yam-egusi.jpg" />
          </div>
          <div className="card-body">
            <div className="card-title">Pounded yam and egusi</div>
            <div className="card-description">
              Sautéed chicken strips in a unique blend of basil pesto, Grana Padano and toasted pine nuts served over fusilli pasta with fresh cream drizzle
            </div>
          </div>
          <div className="card-footer">
            <span>
              <b>Price</b>: &#8358;1,500
            </span>
            <div className="card-options">
              <a className="order-btn" href="login.html">Order</a>
            </div>
          </div>
        </div>
        <div className="card">
          <div className="card-img">
            <img src="img/jollof-rice.jpg" />
          </div>
          <div className="card-body">
            <div className="card-title">Jollof rice and chicken</div>
            <div className="card-description">
              Sautéed chicken strips in a unique blend of basil pesto, Grana Padano and toasted pine nuts served over fusilli pasta with fresh cream drizzle
            </div>
          </div>
          <div className="card-footer">
            <span>
              <b>Price</b>: &#8358;1,500
            </span>
            <div className="card-options">
              <a className="order-btn" href="login.html">Order</a>
            </div>
          </div>
        </div>
        <div className="card">
          <div className="card-img">
            <img src="img/fried-rice.jpg" />
          </div>
          <div className="card-body">
            <div className="card-title">Fried rice and chicken</div>
            <div className="card-description">
              Sautéed chicken strips in a unique blend of basil pesto, Grana Padano and toasted pine nuts served over fusilli pasta with fresh cream drizzle
            </div>
          </div>
          <div className="card-footer">
            <span>
              <b>Price</b>: &#8358;1,500
            </span>
            <div className="card-options">
              <a className="order-btn" href="login.html">Order</a>
            </div>
          </div>
        </div>
        <div className="card">
          <div className="card-img">
            <img src="img/yam-porridge.jpg" />
          </div>
          <div className="card-body">
            <div className="card-title">Yam porridge</div>
            <div className="card-description">
              Sautéed chicken strips in a unique blend of basil pesto, Grana Padano and toasted pine nuts served over fusilli pasta with fresh cream drizzle
            </div>
          </div>
          <div className="card-footer">
            <span>
              <b>Price</b>: &#8358;1,500
            </span>
            <div className="card-options">
              <a className="order-btn" href="login.html">Order</a>
            </div>
          </div>
        </div>
    </div>
    <script type="text/javascript" src="js/index.js"></script>
  </div>
  <footer>
    &copy; 2018 Book-A-Meal Inc.
  </footer>
  </div>
)

export default Home;
