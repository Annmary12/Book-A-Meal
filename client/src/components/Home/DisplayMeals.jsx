import React from 'react';
import PropTypes from 'prop-types';
import MealCard from './MealCard.jsx';
import NoMeal from './NoMeal.jsx';

const DisplayMeals = (props) => {
  if (props.meals.length > 0) {
    return (
      <div className="meals">
        {props.meals.map(meal => (
          <MealCard
            key={meal.id}
            image={meal.img}
            description={meal.description}
            title={meal.title}
            price={meal.price}
          />))
        }
      </div>
    );
  }
  return <NoMeal />;
};

DisplayMeals.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default DisplayMeals;
