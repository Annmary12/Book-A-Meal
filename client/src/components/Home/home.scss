.hungry{
  font-style: normal !important;
  margin: 3px;
  cursor: pointer;
  transition: font-size 1s
  
}

.hungry.bounceIn{
  font-size: 40px;
}