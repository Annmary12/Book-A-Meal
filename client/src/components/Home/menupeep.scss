.peep{
  flex: 1;
  width: 90%;
  margin: auto;
  
  @media only screen and (max-width: 600px){
    display: flex;
		flex-direction: column;
  }
}

.heading{
  text-align: center;
  font-size: 25px;
  padding-top: 15px;
  padding-bottom: 20px;


  @media only screen and (max-width: 600px){
    text-align: center;
		margin-bottom: 19px;
		font-size: 19px;
  }
}