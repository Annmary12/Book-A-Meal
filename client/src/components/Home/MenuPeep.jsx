import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DisplayMeals from './DisplayMeals.jsx';
import LoadingMeals from './LoadingMeals.jsx';
import styles from './menupeep.scss';

/**
 * @class MenuPeep
 */
class MenuPeep extends Component {
  /**
   * @returns {undefined} - undefined
   */
  componentDidMount() {
    this.props.getMenus();
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    return (
      <div className={styles.peep}>
        <div className={styles.heading}>
          <span>~~ <b>Peep into today&#39;s menus</b> ~~</span>
        </div>
        { this.props.loading ? <LoadingMeals /> : <DisplayMeals meals={this.props.meals} /> }
      </div>
    );
  }
}


MenuPeep.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  getMenus: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};

export default MenuPeep;
