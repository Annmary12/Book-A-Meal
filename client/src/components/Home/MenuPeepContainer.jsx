import React, { Component } from 'react';
import axios from 'axios';
import MenuPeep from './MenuPeep.jsx';

/**
 * @class MenuPeepContainer
 */
class MenuPeepContainer extends Component {
  state = {
    loading: false,
    meals: []
  };

  /**
   * @returns {undefined} - undefined
   */
  componentDidMount() {
    this.state.loading = true;

    axios.get(`${APP_URL}/api/v1/menu`) // eslint-disable-line no-undef
      .then((res) => {
        const { menus } = res.data;
        const meals = menus.reduce((todaysMeals, menu) =>
          todaysMeals.concat(menu.categories.reduce((menuMeals, category) =>
            menuMeals.concat(category.meals), [])), []);

        this.state.meals = meals;
        this.state.loading = false;
      })
      .catch((err) => {
        this.state.loading = false;
        console.log(err);
      });
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    return <MenuPeep {...this.state} />;
  }
}

export default MenuPeepContainer;
