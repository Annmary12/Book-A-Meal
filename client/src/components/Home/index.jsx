import React, { Fragment } from 'react';
import Welcome from './Welcome.jsx';
import Footer from '../layout/Footer.jsx';
import MenuPeepContainer from './MenuPeepContainer.jsx';
import '../../../assets/css/style.css';
import '../../../assets/css/index.css';

const Home = () => (
  <Fragment>
    <Welcome />
    <MenuPeepContainer />
    <Footer />
  </Fragment>
);

export default Home;
