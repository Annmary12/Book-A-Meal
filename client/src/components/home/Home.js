import React, { Fragment } from 'react';
import Welcome from './Welcome/Welcome.js';
import Footer from '../Layout/Footer/Footer.js';
import MenuPeepContainer from './MenuPeep/MenuPeepContainer.js';
// import '../../../assets/css/index.css';
// import '../../../assets/css/style.css';

const Home = () => (
  <Fragment>
    <div style={{ flex: 1 }}>
      <Welcome />
      <MenuPeepContainer />
    </div>
    <Footer />
  </Fragment>
);

export default Home;
