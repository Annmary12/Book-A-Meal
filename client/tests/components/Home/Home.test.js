import React from 'react';
import Home from '../../../src/components/Home/Home';

describe('Home component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  })
})