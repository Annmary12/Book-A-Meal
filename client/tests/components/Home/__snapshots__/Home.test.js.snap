// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
<Fragment>
  <div
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <div>
      <Welcome />
      <Connect(MenuPeep) />
    </div>
  </div>
  <Footer />
</Fragment>
`;
